### ISSUE TITLE
Native code library failed to load.

### ISSUE THREAD

**Issue by flocela** [KhronosGroup/Vulkan-Samples#812]
I'm trying to get the vulcan samples to show on my android emulator.
The emulator starts, then a pop up shows "Native code library failed to load."

This is the output from the Run window:

2023-09-22 17:12:09: Launching app on 'Pixel 7 Pro API UpsideDownCakePrivacySandbox.
$ adb shell am start -n "com.khronos.vulkan_samples/com.khronos.vulkan_samples.SampleLauncherActivity" -a android.intent.action.MAIN -c android.intent.category.LAUNCHER --splashscreen-show-icon 

Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=com.khronos.vulkan_samples/.SampleLauncherActivity }

Open logcat panel for emulator Pixel 7 Pro API UpsideDownCakePrivacySandbox
Connected to process 8747 on device 'Pixel_7_Pro_API_UpsideDownCakePrivacySandbox [emulator-5554]'.

===========================================================================================

I've attached the error log from Logcat as errorlog.txt
[errorlog.txt](https://github.com/KhronosGroup/Vulkan-Samples/files/12705190/errorlog.txt)

I'm new to android and Vulkan. Your help is appreciated.

**Comment by SaschaWillems** [comment]
Please see #764 and check if that fixes your problem.

**Comment by flocela** [comment]
I'm sorry to be such a problem. My apk files look much line SZU-WenjieHuang's in #764. I too only have libc++_shared.so and libVkLayer_khronos_validation.so under the  libs/"ABIname" directories. I'm not sure if that's the problem. I tried his solution of generating the gradle project again. In the top directory (Vulkan-Samples which contains the build directory) I run ./scripts/generate.py android

Then I go into build/gradle by typing cd build/android_gradle

Then I type gradle assembleDebug.

That gives me this error:
```
* Where:
Build file '/Users/xxx/Vulkan-Samples/build/android_gradle/app/build.gradle' line: 2

* What went wrong:
An exception occurred applying plugin request [id: 'com.android.application']
> Failed to apply plugin 'com.android.internal.application'.
   > Failed to create service 'com.android.build.gradle.options.ProjectOptionService_fd0cf32d-cde6-4deb-85f0-60446a692c1c'.
      > Could not create an instance of type com.android.build.gradle.options.ProjectOptionService.
         > Cannot obtain value from provider of Gradle property 'android.testInstrumentationRunnerArguments.package' at configuration time.
           Use a provider returned by 'forUseAtConfigurationTime()' instead.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
```
I've included my app/build.gradle file.

[build.gradle.txt](https://github.com/KhronosGroup/Vulkan-Samples/files/12721039/build.gradle.txt)

If I then regenerate the Android gradle files with the ./scripts/generate.py android command. Then I use Android Studio to Build the APK's (By pressing Build > Build Bundles > Build APK's), then it builds successfully. The apk does have libVkLayer_khronos_validation.so and libc++_shared.so in the lib/"ABI-names" directories.

![Screen Shot 2023-09-25 at 6 08 52 PM](https://github.com/KhronosGroup/Vulkan-Samples/assets/4298622/0b1ea03a-2ed7-4364-b90c-5e0f56a359fe)

I press run and I get the same Native code library failed to load. popup in the AVD.

I've been running with the "Deploy: APK from app bundle" run configuration.
![DeployFromAppBundle](https://github.com/KhronosGroup/Vulkan-Samples/assets/4298622/16489bef-39b6-425b-99b0-9d28710fba9f)

Thank you very much!
